src/Nethermind/Nethermind.Facade/Simulate/SimulateTransactionProcessor.cs

```diff
@@ -20,13 +21,24 @@ public class SimulateTransactionProcessor(
     bool validate)
     : TransactionProcessor(specProvider, worldState, virtualMachine, codeInfoRepository, logManager), ITransactionProcessor
 {
+    public UInt256? BlobBaseFee { get; set; }
```

Comment:
Why can't it be passed in BlockExecutionContext already here?

Reply:
@LukaszRozmej It seems to me the only place where we could control BEC would be in `SimulateBlockValidationTransactionsExecutor`. To implement this change we would need:
 - only use `SimulateBlockValidationTransactionsExecutor` in simulate (see `GetProcessor` in `SimulateBlockValidationTransactionsExecutor`) and bring "`bool validate`" into it
 - `SimulateBlockValidationTransactionsExecutor` will need to overload `BlockValidationTransactionsExecutor.ProcessTransactions` at line 39 

which is rather complicated in comparison to the current approach. Yet we can do it if it is better. Is it?

Reply:
Yep pushed it myself